.github/workflows/rust-next.yml

```diff
@@ -0,0 +1,240 @@
+
+      # We can use the registry cache of the normal rust ci
+      - name: Cache key
+        id: cache-key
+        run: echo "::set-output name=key::$(echo ${{ runner.os }}-cargo-registry-${{ steps.get-date.outputs.date }}-${{ hashFiles('**/Cargo.lock') }})"
+        shell: bash
+
+      - name: Cache cargo registry
+        uses: actions/cache@v2.1.4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ steps.cache-key.outputs.key }}
+          restore-keys: ${{ runner.os }}-cargo-registry-${{ steps.get-date.outputs.date }}-
+
+      - name: cargo fetch
+        working-directory: ./rust
+        run: cargo fetch
+
+  format:
+    name: cargo-fmt
+    needs: registry-cache
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        cargo_manifest: [rust, bindings/python]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install nightly toolchain
+        id: rust-toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt
+          default: true
+
+      - name: Use cached cargo registry
+        uses: actions/cache@v2.1.4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ needs.registry-cache.outputs.cache-key }}
+
+      # cargo fmt does not create any artifacts, therefore we don't need to cache the target folder
+
+      - name: cargo fmt
+        working-directory: ${{ matrix.cargo_manifest }}
+        run: cargo fmt --all -- --check
+
+  check:
+    name: cargo-check
+    needs: registry-cache
+    timeout-minutes: 20
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        rust_version: [stable, beta]
+        cargo_manifest: [rust, bindings/python]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        id: rust-toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust_version }}
+          default: true
+
+      - name: Use cached cargo registry
+        uses: actions/cache@v2.1.4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ needs.registry-cache.outputs.cache-key }}
+
+      - name: Cache build artifacts
+        uses: actions/cache@v2.1.4
+        with:
+          path: ${{ matrix.cargo_manifest }}/target
+          key: ${{ runner.os }}-cargo-${{ steps.rust-toolchain.outputs.rustc }}-check-${{ matrix.cargo_manifest }}-${{ needs.registry-cache.outputs.cache-date }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-${{ steps.rust-toolchain.outputs.rustc }}-${{ matrix.cargo_manifest }}-check-${{ needs.registry-cache.outputs.cache-date }}-
```

Comment:
I think we can remove the cache for the build artifacts
[because caches that are not accessed within a week will be evicted](https://github.com/actions/cache#cache-limits)
```suggestion
```